### PR TITLE
doc/dev: improve Basic Workflow wording

### PR DESCRIPTION
**Pull request by zdover23**
Make several semantic and syntactic improvements to basic-workflow.rst.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/dev/developer_guide/basic-workflow.rst [hunk 1]
```diff
@@ -184,9 +184,8 @@ Because upstream main is continually receiving updates from other
 contributors, your fork will drift farther and farther from the state of the
 upstream repo when you cloned it.
 
-You must keep your fork's main branch synchronized with upstream main in
-order to reduce drift between your fork's main branch and the upstream main
-branch.
+Keep your fork's main branch synchronized with upstream main to reduce drift
+between your fork's main branch and the upstream main branch.
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/basic-workflow.rst:188]: Backticks `main`?

**Review by anthonyeleven — APPROVED** [review]

File: doc/dev/developer_guide/basic-workflow.rst (hunk 1 above)
Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:188]: Good call.

Comment (anthonyeleven) [reply doc/dev/developer_guide/basic-workflow.rst:188]: I loves me some markup

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/49077 - Quincy backport
https://github.com/ceph/ceph/pull/49078 - Pacific backport

**Pull request merged**